src/gui/ScreenHandler.cpp

```diff
@@ -163,7 +173,35 @@ void Screens::PushBeforeCurrent(const ScreenFactory::Creator screen_creator) {
     }
 }
 
+void Screens::ResetTimeout() {
+    if (GuiDefaults::menu_timeout_enabled) {
+        if (gui_get_menu_timeout_id() >= 0) {
```

Comment:
please double or triple check gui_timers before use
they were broken and I am not sure if they are fixed

in this case I would just check time stamps instead of use complicated gui_timers

Reply:
They are working just fine. I had a look on them, made a little debug and did not see any potential issue.